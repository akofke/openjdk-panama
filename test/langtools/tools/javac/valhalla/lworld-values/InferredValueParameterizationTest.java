/*
 * @test /nodynamiccopyright/
 * @bug 8210346
 * @summary inferred value typed `type arguments' are allowed by Javac even without -XDallowGenericsOverValues
 * @compile/fail/ref=InferredValueParameterizationTest.out -XDrawDiagnostics -XDdev InferredValueParameterizationTest.java
 * @compile -XDallowGenericsOverValues -XDrawDiagnostics -XDdev InferredValueParameterizationTest.java
 */

import java.util.List;

public value class InferredValueParameterizationTest {
    int x = 10;

    static class Y<T> {
        Y(T t) {}
    }

    static <K> List<K> foo(K k) {
        return null;
    }

    public static void main(String [] args) {
       var list = List.of(new InferredValueParameterizationTest());
       Object o = new Y<>(new InferredValueParameterizationTest());
       o = new Y<>(new InferredValueParameterizationTest()) {};
       foo(new InferredValueParameterizationTest());
    }
}
