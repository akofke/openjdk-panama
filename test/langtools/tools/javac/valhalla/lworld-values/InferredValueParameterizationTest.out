InferredValueParameterizationTest.java:23:26: compiler.err.generic.parameterization.with.value.type: java.util.List<InferredValueParameterizationTest>
InferredValueParameterizationTest.java:23:12: compiler.err.generic.parameterization.with.value.type: java.util.List<InferredValueParameterizationTest>
InferredValueParameterizationTest.java:24:19: compiler.err.generic.parameterization.with.value.type: InferredValueParameterizationTest.Y<InferredValueParameterizationTest>
InferredValueParameterizationTest.java:25:12: compiler.err.generic.parameterization.with.value.type: InferredValueParameterizationTest.Y<InferredValueParameterizationTest>
InferredValueParameterizationTest.java:26:11: compiler.err.generic.parameterization.with.value.type: java.util.List<InferredValueParameterizationTest>
5 errors
